### DISCUSSION TITLE
Enable `tsconfig` setting `noImplicitOverride`

### DISCUSSION THREAD

**Discussion by MrChocolatine** [typed-ember/ember-cli-typescript#1458] [Ideas]
As a `ember-cli-typescript` user, I would like the `tsconfig` setting `noImplicitOverride` to be enabled so that TypeScript warns me whenever an overriden method from a parent class gets out of sync.

I really think this setting can help having a more robust codebase.

More information:
- https://www.typescriptlang.org/tsconfig#noImplicitOverride
- https://devblogs.microsoft.com/typescript/announcing-typescript-4-3/#override

> When working with classes which use inheritance, it’s possible for a
> sub-class to get “out of sync” with the functions it overloads when
> they are renamed in the base class.
> ...
> Using `noImplicitOverride` you can ensure that the sub-classes never
> go out of sync, by ensuring that functions which override include the
> keyword `override`.

**Comment by chriskrycho** [comment]
Agreed! The simple fix here will actually be to stop setting per-strictness-flag values and use `strict: true`.

**Comment by MrChocolatine** [comment]
> Greed! The simple fix here will actually be to stop setting per-strictness-flag values and use `strict: true`.

Completely agree with that!

But in this case, the option `noImplicitOverride` is not automatically enabled when having `strict: true`.

**Comment by chriskrycho** [comment]
Ah, I forgot. That's one of their "lint"-type rules. We probably need a small RFC to define which lint rules we should opt into by default (if any), vs. allowing other tools (like `@typescript-eslint`) to fix. One big reason for that: we currently default to `noEmitOnError: true`, so I think it's fairly important that as long as preserve that default, we *don't* block compilation for things which are lint-style errors rather than type errors in the "correctness" sense.

**Comment by MrChocolatine** [comment]
> we _don't_ block compilation for things which are lint-style errors rather than type errors in the "correctness" sense.

Okay, I get that, makes sense.

Then perhaps, we could have a new page in the documentation about all these "lint"-type setting, such as `noImplicitOverride` here, that are interesting but not enabled by default? (or simply updating the page https://docs.ember-cli-typescript.com/ts/using-ts-effectively)

**Comment by chriskrycho** [comment]
While I definitely appreciate the suggestion, I think that’s all TypeScript-specific information and should live in *TypeScript*’s docs. If it’s not clear from the TS compiler options guide or handbook (and I agree that it isn’t!) we should work to address it there. Happily, as with TS itself, the docs are open source, and I know they’re quite receptive to feedback on stuff like this.

(Note: I’m going to convert this to a Discussion.)
